### ISSUE TITLE
Navigation menu block: consider the block navigator as an alternative interface to add items

### ISSUE THREAD

**Issue by enriquesanchez** [WordPress/gutenberg#20742]
As part of the accessibility test performed by the Accessibility team (#20369) we found that keyboard-only navigation in the navigation menu block was lacking.

On our testing, we found the block navigator UI easier to interact with:

> [with keyboard-only] I was able to add a submenu item by way of the 'Add submenu' option in the block's toolbar. This pattern feels more intuitive and easier to navigate. However, having the interaction split into two different places (on the modal I create a new navigation link, but then I have to go back to the block to edit the newly added link placeholder) was confusing.

![2020-03-09 14-33-16 2020-03-09 14_35_44](https://user-images.githubusercontent.com/3276087/76255145-4d5c0b80-6213-11ea-876a-7d02a83d8bbd.gif)

Consider exploring more ideas around this pattern and allowing the user to fully create a navigation menu, without the need to jump between two different UIs, which if you are a keyboard-only user, is hard.

**Comment by mtias** [comment]
Are there specific things to add to the block navigator? That was always the plan to make it a first-class interface. There's a bunch of issues with tasks for it.

**Comment by enriquesanchez** [comment]
I think that being able edit new menu tiems (name, URL, etc) from the same UI would go a long way. Right now I can only create a new link, and I have to then go back to the block to edit it.

**Comment by mtias** [comment]
Would you mind creating an issue for that specific task? It'd be easier to track that way.

**Comment by mtias** [comment]
I agree with this approach, but let's close this issue as there is nothing really actionable here and open specific issues for any missing features or gaps.
